gbn/gbn_conn.go

```diff
@@ -582,6 +788,13 @@ func (g *GoBackNConn) receivePacketsForever() error { // nolint:gocyclo
 			if gotValidACK {
 				g.resendTicker.Reset(g.resendTimeout)
 
+				g.awaitingCatchUpMu.RLock()
```

Comment:
it's scary to me to hold a lock during a time.Sleep - if we have a subsystem handling these things rather, then it will make it easier to just swallow things during the wait

Reply:
See https://github.com/lightninglabs/lightning-node-connect/pull/87#discussion_r1391462799 (1)